### DISCUSSION TITLE
Calcul de distances entre deux CAO (deux Mesh3D/Shell)

### DISCUSSION THREAD

**Discussion by MackJoly** [Dessia-tech/volmdlr#1334] [General]
Bonjour l'équipe,
J'aimerai calculer (précisément) la distance entre deux CAO (de type Mesh3D ou Shell).
Je vois qu'il y a une méthode dans Mesh3D qui le fait mais en utilisant seulement les arêtes des triangles il me semble (minimum_distance).
Existe-t-il d'autres méthodes que je n'aurai pas vu ou à implémenter ?

**Comment by WirajanDASILVA** [comment]
Salut, Mack!! il y avait une autre méthode, mais a été remplacé par celui de Mesh3D. Les résultats de cette méthode  ne sont pas assez précis?

**Comment by pgibertini** [comment]
Bonjour Mack,

Pour le moment il n’y a pas de méthode exacte. La méthode approchée (`minimum_distance`) de `Mesh3D` utilise les sommets des triangles (je crois). 

Cette méthode est rapide est relativement précise selon ton maillage. Nous n’avons pas encore pris le temps de développer une méthode exacte. 
Si nécessaire, tu peux faire un ticket pour une méthode exact.

**Comment by MackJoly** [comment]
Je sais qu'il existe aussi dans Shell le minimum_distance
Je vais comparer les précisions mais comme le mentionne Pierre, la méthode de Mesh3D utilise que les sommets et j'ai des CAO avec de grands triangles (donc pas précis)
Je voulais aussi savoir si vous aviez des idées de méthodes
Je ferai un ticket, je pense que ça peut être intéressant de développer des algos pour précis

**Comment by pgibertini** [comment]
Il y avait une autre méthode supposée exacte, sauf qu’elle donnait des valeurs de distance minimale plus grande que la méthode utilisant les points, ce qui n’est pas normal (à confirmer).

Je te laisse essayer avec les méthodes sur les triangle shell.

**Comment by MackJoly** [comment]
Je reviens vers vous après comparaison rapide de mon côté sur le statut des deux méthodes (Mesh3D et Shell)

**Comment by MackJoly** [comment]
Re ! Pour info, j'ai lancé un test de perf et le résultat est sans appel, la méthode de Mesh3D est plus performante et semble donner un résultat similaire (à valider sur plusieurs use cases)

```
Shell to shell: 37.14541486811069 mm. 18618.780987024307 seconds -> plus de 5h
Mesh to mesh: 37.14541486811069 mm 1.2098538875579834 seconds
```

**Comment by MackJoly** [comment]
J'ai testé une méthode custom, voici le screen résultat:
![image](https://github.com/Dessia-tech/volmdlr/assets/61141035/f916e507-22a0-49bf-8e92-5ae75fc85c81)
La face du bas est un grand triangle (grande surface plane)
L'algo Mesh3D.distance prend 1.2 secondes, ma méthode custom prend 1.7 secondes
 je ne sais pas me positionner sur la performance entre 2 grosses CAO (à tester)
Ici le test s'est fait avec une grosse CAO (faces rouges) et une plus petite (tube orange)
